DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/Delegates/AllDelegates/AllDelegatesViewModelTests.cs

```diff
@@ -0,0 +1,317 @@
+                { FirstName = "d purple", LastName = "Surname" },
+            new DelegateUserCard
+                { FirstName = "e", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "f", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "g", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "h", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "i", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "j", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "k", LastName = "Surname", DateRegistered = DateTime.Today },
+            new DelegateUserCard
+                { FirstName = "l", LastName = "Surname", Answer4 = "C 1" },
+            new DelegateUserCard
+                { FirstName = "m", LastName = "Surname", Answer4 = "C 2" },
+            new DelegateUserCard
+                { FirstName = "n", LastName = "Surname", Answer4 = "C 2" },
+            new DelegateUserCard
+                { FirstName = "o", LastName = "Surname", DateRegistered = DateTime.Today.AddDays(1) }
+        };
+
+        public AllDelegatesViewModelTests()
+        {
+            customPromptsService = A.Fake<ICentreCustomPromptsService>();
+            customPromptHelper = new CentreCustomPromptHelper(customPromptsService);
+        }
+
+        [Test]
+        public void All_delegates_should_default_to_returning_the_first_ten_delegates()
+        {
+            var model = new AllDelegatesViewModel(
+                delegateUsers,
+                new List<(int, string)>(),
+                new List<CustomPrompt>(),
+                1,
+                null,
+                DelegateSortByOptions.Name.PropertyName,
+                BaseSearchablePageViewModel.Ascending,
+                null
+            );
+
+            model.Delegates.Count().Should().Be(10);
+            model.Delegates.FirstOrDefault(delegateUser => delegateUser.DelegateInfo.Name == "k Surname").Should()
+                .BeNull();
+        }
+
+        [Test]
+        public void All_delegates_should_correctly_return_the_second_page_of_delegates()
+        {
+            var model = new AllDelegatesViewModel(
+                delegateUsers,
+                new List<(int, string)>(),
+                new List<CustomPrompt>(),
+                2,
+                null,
+                DelegateSortByOptions.Name.PropertyName,
+                BaseSearchablePageViewModel.Ascending,
+                null
+            );
+
+            model.Delegates.Count().Should().Be(5);
+            model.Delegates.First().DelegateInfo.Name.Should().BeEquivalentTo("k Surname");
+        }
+
+        [Test]
+        public void All_delegates_filters_should_be_set()

```

Comment:
This test is more than 100 lines long, which means that it is very difficult to just scan over - could we try and refactor it to make it easier to read? :)